### ISSUE TITLE
Teleport Application Access - CORS / Native REST communication to another Teleport Application App. 

### ISSUE THREAD

**Issue by benarent** [gravitational/teleport#8502]
### Description

**What happened**:

I've been working with the LFEDGE Sig team on getting Fledge to work with  with Teleport Application Access. 

This is the [LF-Edge Fledge POC](https://wiki.lfedge.org/display/OH/Milestone+1%3A+Table+Garden)

I'm able to connect both apps.

![image](https://user-images.githubusercontent.com/559288/136289926-00d1042c-f6e9-4448-b18e-2392511f7d5c.png)

I can connect to /flege/ping on the pi via the browser and can curl the endpoint using certs.
```
curl \
  --cacert /Users/benarent/.tsh/keys/teleport-7.asteroid.earth/certs.pem \
  --cert /Users/benarent/.tsh/keys/teleport-7.asteroid.earth/benarent-app/teleport-7.asteroid.earth/fledgepi-x509.pem \
  --key /Users/benarent/.tsh/keys/teleport-7.asteroid.earth/benarent \
  https://fledgepi.teleport-7.asteroid.earth:443/fledge/ping
{"uptime": 578, "dataRead": 0, "dataSent": 0, "dataPurged": 0, "authenticationOptional": true, "serviceName": "Fledge", "hostName": "raspberrypi", "ipAddresses": ["192.168.0.126", "192.168.0.127", "fdec:e08a:7ad2:1:5837:c275:90aa:6d34"], "health": "green", "safeMode": false, "version": "1.9.2"}%
```

I can access both apps, but I'm hitting two issues. I'm unable to connect the 'FledgeGUI -> Fledge Data services` on the Pi, since it's making a local HTTP request over the browser. 

1. CORS issue <img width="1555" alt="Screen Shot 2021-10-05 at 10 08 42 PM" src="https://user-images.githubusercontent.com/559288/136290067-0134439b-39e2-47e2-ba7c-2b7dbfc30dc0.png">
2. Response Redirect Auth Issue. <img width="739" alt="Screen Shot 2021-10-06 at 2 32 30 PM" src="https://user-images.githubusercontent.com/559288/136290003-6cedcef4-2e86-4a8f-8903-3b468fc078a8.png">

Debugged with @r0mant with a custom 7.2 test build.

**Comment by jonaskello** [comment]
I have this exact problem with a SPA that has a separate login server. I can access both the SPA and the login server. However when the SPA starts it will make a request to fetch the openid-configuration and teleport will respond with a 302 and the launch url as a new location. At this point the SPA fails with a CORS error.

So the scenario is this

SPA app at: `https://myapp.teleport.mydomain.se/`

Login server at: `https://login-server.teleport.mydomain.se/`

The SPA app makes a call to 

`https://login-server.teleport.mydomain.se/auth/realms/myapp/.well-known/openid-configuration` 

And the response is `status 302`, `location: https://teleport.mydomain.se:443/web/launch/login-server.teleport.divid.se`

And the error in javascript console is:

```
Access to XMLHttpRequest at 'https://login-server.teleport.mydomain.se/auth/realms/myapp/.well-known/openid-configuration' from origin 'https://myapp.teleport.divid.se' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

oidc-client.min.js:1 GET https://login-server.teleport.divid.se/auth/realms/myapp/.well-known/openid-configuration net::ERR_FAILED 302
```

**Comment by jonaskello** [comment]
I think perhaps this could be solved if teleport would allow you to configure CORS for each site that is exposed as an application?

**Comment by ridwanhady** [comment]
Hi! I face the same problem with this, any workaround yet?

**Comment by webvictim** [comment]
I'm having this same issue with the Unifi console application.

**Comment by aqiank** [comment]
I'm also having this problem. Is there any plan to support CORS? Right now, I have to compromise by allowing client access to the API without going through Teleport authentication. This seems like something that should be supported.

**Comment by webvictim** [comment]
This is a similar request to https://github.com/gravitational/teleport/issues/17588

**Comment by zmb3** [comment]
See also #8307

**Comment by moisesmejia1200** [comment]
During our product sync this week we talked about this but it was requested to get more information. Kindly find here

https://drive.google.com/drive/folders/1FEbNLYd0PiBXJrXAMi-xLCVNer2C0XJP?usp=sharing

**Comment by zmb3** [comment]
Tracking this over at #17588

**Comment by benarent** [comment]
This issue has been fixed with Teleport 16.4.0 with Backends-for-Frontends support  & CORS support for preflight requests.

https://goteleport.com/docs/enroll-resources/application-access/guides/connecting-apps/#backends-for-frontends-support
